test/test_e2e_asr_mulenc.py

```diff
@@ -0,0 +1,407 @@
+        ctc_window_margin=0,
+        lm_weight=0.0,
+        rnnlm=None,
+        streaming_min_blank_dur=10,
+        streaming_onset_margin=2,
+        streaming_offset_margin=2,
+        verbose=2,
+        char_list=[u"あ", u"い", u"う", u"え", u"お"],
+        outdir=None,
+        ctc_type="warpctc",
+        report_cer=False,
+        report_wer=False,
+        sym_space="<space>",
+        sym_blank="<blank>",
+        sortagrad=0,
+        grad_noise=False,
+        context_residual=False,
+        use_frontend=False,
+        replace_sos=False,
+        tgt_lang=False,
+        share_ctc=False,
+        weights_ctc_train=[0.5, 0.5],
+        weights_ctc_dec=[0.5, 0.5],
+    )
+    defaults.update(kwargs)
+    return argparse.Namespace(**defaults)
+
+
+def prepare_inputs(mode, ilens_list=[[20, 15], [19, 16]], olens=[4, 3], is_cuda=False):
+    np.random.seed(1)
+    assert len(ilens_list[0]) == len(ilens_list[1]) == len(olens)
+    xs_list = [[np.random.randn(ilen, 40).astype(np.float32) for ilen in ilens] for ilens in ilens_list]
+    ys = [np.random.randint(1, 5, olen).astype(np.int32) for olen in olens]
+    ilens_list = [np.array([x.shape[0] for x in xs], dtype=np.int32) for xs in xs_list]
+
+    if mode == "pytorch":
+        ilens_list = [torch.from_numpy(ilens).long() for ilens in ilens_list]
+        xs_pad_list = [pad_list([torch.from_numpy(x).float() for x in xs], 0) for xs in xs_list]
+        ys_pad = pad_list([torch.from_numpy(y).long() for y in ys], -1)
+        if is_cuda:
+            xs_pad_list = [xs_pad.cuda() for xs_pad in xs_pad_list]
+            ilens_list = [ilens.cuda() for ilens in ilens_list]
+            ys_pad = ys_pad.cuda()
+
+        return xs_pad_list, ilens_list, ys_pad
+    else:
+        raise ValueError("Invalid mode")
+
+
+def convert_batch(batch, backend="pytorch", is_cuda=False, idim=40, odim=5, num_inputs=2):
+    ilens_list = [np.array([x[1]['input'][idx]['shape'][0] for x in batch]) for idx in range(num_inputs)]
+    olens = np.array([x[1]['output'][0]['shape'][0] for x in batch])
+    xs_list = [[np.random.randn(ilen, idim).astype(np.float32) for ilen in ilens_list[idx]] for idx in
+               range(num_inputs)]
+    ys = [np.random.randint(1, odim, olen).astype(np.int32) for olen in olens]
+    is_pytorch = backend == "pytorch"
+    if is_pytorch:
+        xs_list = [pad_list([torch.from_numpy(x).float() for x in xs_list[idx]], 0) for idx in range(num_inputs)]
+        ilens_list = [torch.from_numpy(ilens_list[idx]).long() for idx in range(num_inputs)]
+        ys = pad_list([torch.from_numpy(y).long() for y in ys], -1)
+
+        if is_cuda:
+            xs_list = [xs_list[idx].cuda() for idx in range(num_inputs)]
+            ilens_list = [ilens_list[idx].cuda() for idx in range(num_inputs)]
+            ys = ys.cuda()
+
+    return xs_list, ilens_list, ys
+
+
+@pytest.mark.parametrize(
+    "module, model_dict", [
+        ('espnet.nets.pytorch_backend.e2e_asr_mulenc', {}),
```

Comment:
if this mulenc can work with `num_encs > 2`, it is better to add test case `('espnet.nets.pytorch_backend.e2e_asr_mulenc', {'num_encs': 3}),`

Reply:
Thanks, I have added test case for num_encs = 3.